### PR TITLE
Contact Form: add deprecation warnings to Contact Form module codebase

### LINKED ISSUE #35654: Contact Form: add deprecation warnings
In the module codebase, add deprecation warnings (`_deprecated_function`) in the hook callbacks and class methods (make sure they appear in the changelog too).

Project thread: pf5801-Aj-p2

### PR DESCRIPTION
**Pull request by monsieur-z**
Fixes https://github.com/Automattic/jetpack/issues/35654

## Proposed changes:

Add deprecation warnings to the global functions, classes, public methods, and properties in the Contact Form module codebase. The code has been brought over to the Forms package.

This PR also updates some Jetpack plugin tests to reference the Forms package instead of the Contact Form module. Indeed, the deprecation warnings added to the module made many tests fail.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Project thread: pf5801-Aj-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

There's not much to test here.

- Spin up a test site with this branch
- Make sure you have access to the server logs
- Use the Contact Form module instead of the Forms package by ensuring the condition at this line is falsy: https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/contact-form.php#L33
- Verify you see the deprecation warnings in the logs

### CODE REVIEW

File: projects/plugins/jetpack/tests/php/modules/contact-form/test-class.grunion-contact-form.php [hunk 1]
```diff
@@ -975,6 +1000,43 @@ public function assertCommonValidHtml( $wrapper_div, $attributes ) {
 		$this->assertEquals( trim( $label->nodeValue ), $attributes['label'], 'Label is not what we expect it to be...' );
 	}
 
+	/**
+	 * Tests whether the class attribute in the wrapper div matches the field's class attribute value.
+	 *
+	 * @param DOMElement $wrapper_div The wrapper div.
+	 * @param array      $attributes An associative array containing the field's attributes.
+	 */
+	public function assertFieldClasses( $wrapper_div, $attributes ) {
```
Comment (monsieur-z) [inline projects/plugins/jetpack/tests/php/modules/contact-form/test-class.grunion-contact-form.php:1009]: Copied from `/test-class.contact-form.php`. Same for all updated asserts function in this file.

File: projects/plugins/jetpack/modules/contact-form/admin.php [hunk 2]
```diff
@@ -19,9 +20,12 @@
 /**
  * Build contact form button.
  *
+ * @deprecated $$next-version$$ Use Automattic\Jetpack\Forms\ContactForm\
  * @return void
  */
 function grunion_media_button() {
+	_deprecated_function( __FUNCTION__, '$$next-version$$', 'Automattic\Jetpack\Forms\ContactForm\Admin->' );
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/contact-form/admin.php]: Looks like the method is missing from the replacement function.

Comment (jeherve) [inline projects/plugins/jetpack/modules/contact-form/admin.php]: I think we should add `jetpack-` to all those deprecation notices, as recommended here:
https://github.com/Automattic/jetpack/blob/f0fa2c82dce0cb507b892645e27a9fafd8ab87e0/docs/coding-guidelines.md#L71

This is helpful in error logs, folks know the notices are about the Jetpack plugin, and not about core.

```suggestion
	_deprecated_function( __FUNCTION__, 'jetpack-$$next-version$$', 'Automattic\Jetpack\Forms\ContactForm\Admin->grunion_media_button' );
```

**Review by jeherve — COMMENTED** [review]
I would recommend a small change in the deprecation notices.

File: projects/plugins/jetpack/modules/contact-form/admin.php (hunk 2 above)
Comment (monsieur-z) [reply projects/plugins/jetpack/modules/contact-form/admin.php:27]: Good catch!

Comment (monsieur-z) [reply projects/plugins/jetpack/modules/contact-form/admin.php]: Thanks for the tip!

**Review by jeherve — APPROVED** [review]

**Pull request merged**